### ISSUE TITLE
Exception when trying to show About Dialog with missing localizations

### ISSUE THREAD

**Issue by Areiser** [flutter/flutter#14890]
## Steps to Reproduce

When trying to open an about dialog on my Huawei Y-550 for testing, nothing happens. An exception is logged. The same call is successful on my newer Huawei P9 lite.

The code causing this is simply

```dart
print('here be dragons');
showAboutDialog(
   context: context,
   applicationName: MyLocalizations.of(context).appName,
   applicationVersion: '1.0',
   applicationLegalese: "blabla!",
);
```

## Logs

[flutter_about_bug.txt](https://github.com/flutter/flutter/files/1760044/flutter_about_bug.txt)

## Flutter Doctor
```
[✓] Flutter (on Linux, locale en_US.UTF-8, channel alpha)
    • Flutter version 0.0.21 at /run/media/andreas/Andreas/Programmieren/flutter
    • Framework revision 2e449f06f0 (4 weeks ago), 2018-01-29 14:26:51 -0800
    • Engine revision 6921873c71
    • Tools Dart version 2.0.0-dev.16.0
    • Engine Dart version 2.0.0-edge.da1f52592ef73fe3afa485385cb995b9aec0181a

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /home/andreas/Android/Sdk
    • Android NDK at /home/andreas/Android/Sdk/ndk-bundle
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[✓] Android Studio (version 3.0)
    • Android Studio at /opt/android-studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[✓] IntelliJ IDEA Community Edition (version 2017.3)
    • Flutter plugin version 21.2.3
    • Dart plugin version 173.4548.30

[✓] Connected devices
    • HUAWEI Y550 L01 • b4305293246b • android-arm • Android 4.4.4 (API 19)
```

**Comment by Areiser** [comment]
I just wanted to update this since the error message in the exception changed. I'm now using the latest beta.
The issue still only occurs only on the Huawei Y-550. 

It is not related to localization, the same happens when I use plain strings. 

To check if this has something to do with my custom RoundedButton (also seen in the stack trace) I tried the same in a MaterialButton onPressed and it still does not work.

> Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v0.2.8, on Linux, locale en_US.UTF-8)


```
I/flutter (12796): ══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
I/flutter (12796): The following NoSuchMethodError was thrown while handling a gesture:
I/flutter (12796): The method '???' was called on null.
I/flutter (12796): Receiver: null
I/flutter (12796): Tried calling: ???()
I/flutter (12796): When the exception was thrown, this was the stack:
I/flutter (12796): #0      showDialog (package:flutter/src/material/dialog.dart:0)
I/flutter (12796): #1      showAboutDialog (package:flutter/src/material/about.dart:154)
I/flutter (12796): #2      _MyHomePageState.build.<anonymous closure> (file:///run/media/andreas/Andreas/Programmieren/Flutter_Projects/Idiotentest/lib/main.dart:146)
I/flutter (12796): #3      RoundedButton.build.<anonymous closure> (package:idiotentest/quizinterface/RoundedButton.dart:20)
I/flutter (12796): #4      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:478)
I/flutter (12796): #5      _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:530)
I/flutter (12796): #6      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:102)
I/flutter (12796): #7      TapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:161)
I/flutter (12796): #8      TapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:94)
I/flutter (12796): #9      PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:184)
I/flutter (12796): #10     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:73)
I/flutter (12796): #11     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:101)
I/flutter (12796): #12     _WidgetsFlutterBinding&BindingBase&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:143)
I/flutter (12796): #13     _WidgetsFlutterBinding&BindingBase&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:121)
I/flutter (12796): #14     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:101)
I/flutter (12796): #15     _WidgetsFlutterBinding&BindingBase&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:64)
I/flutter (12796): #16     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:48)
I/flutter (12796): #17     _invoke1 (dart:ui/hooks.dart:134)
I/flutter (12796): #18     _dispatchPointerDataPacket (dart:ui/hooks.dart:91)
I/flutter (12796): Handler: onTap
I/flutter (12796): Recognizer:
I/flutter (12796):   TapGestureRecognizer#f7cb1(debugOwner: GestureDetector, state: possible, won arena, finalPosition:
I/flutter (12796):   Offset(218.2, 512.7), sent tap down)
I/flutter (12796): ════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by Areiser** [comment]
I am sorry, this was actually my fault and internationalization related.
Today I did a bit of digging and I realized I can reproduce this when using other languages. The problem seemed to be that 

`MaterialLocalizations.of(context)`

 is not available (this is called in dialog.dart for modalBarrierDismissLabel). I don't use widget localizations at all (all text is contained in my localizations) so I never used 

```
GlobalMaterialLocalizations.delegate,
GlobalWidgetsLocalizations.delegate,
```
in my App. After adding them, the error was gone. Sorry for the inconvenience, but maybe it would be useful for the future if there was a more precise error message when no Localizations for the langauge are available.
